core/async_job/async_job.ex

```diff
@@ -90,18 +92,50 @@ defmodule AntikytheraCore.AsyncJob do
       remaining_attempts: attempts,
       retry_interval:     Keyword.get(options, :retry_interval, RetryInterval.default()),
       payload:            payload,
+      immediate:          immediate_job?(options),
     ])
   end
 
+  defunp immediate_job?(options :: v[[option]]) :: boolean do
+    case options[:immediate] do
+      true ->
+        debug_assert(
+          [:schedule, :attempts, :retry_interval]
+          |> Enum.map(fn disallowed -> not Keyword.has_key?(options, disallowed) end)
+          |> Enum.all?(),
```

Comment:
- We can use `Enum.all?/2` here.
- Let's return errors by `register/5` when incorrect combinations of options are found, instead of using `debug_assert/2`.

Reply:
Fixed in 0a17fb1cc0e34322e8281898a5ec061cef1ecb01

Reply:
I wrote tests for it.

a7b43194bf769976a81f4e2ee8c754fa255991d3